sample_project/Assets/SampleViewer/Samples/Routing/RouteManager.cs

```diff
@@ -132,6 +147,57 @@ private async void OnLeftClickStart(InputAction.CallbackContext context)
         RebaseRoute();
     }
 
+    private async void OnTouchInputStarted(InputAction.CallbackContext ctx)
+    {
+        if (routing)
+        {
+            DisplayNoteText("Please wait for the routing to finish.");
+            animator.Play("NotificationAnim");
+
+            return;
+        }
+        animator.Play("NotificationAnim_Close");
+        DisplayNoteText("Hold Left Shift + Left Click on the map to begin routing.");
+
+        RaycastHit hit;
+        Ray ray = Camera.main.ScreenPointToRay(touchControls.Touch.TouchPosition.ReadValue<Vector2>());
+
+        if (Physics.Raycast(ray, out hit))
+        {
+            var routeMarker = Instantiate(RouteMarker, hit.point, Quaternion.identity, arcGISMapComponent.transform);
+
+            var geoPosition = HitToGeoPosition(hit);
+
+            var locationComponent = routeMarker.GetComponent<ArcGISLocationComponent>();
+            locationComponent.enabled = true;
+            locationComponent.Position = geoPosition;
+            locationComponent.Rotation = new ArcGISRotation(0, 90, 0);
+
+            stops.Enqueue(routeMarker);
+
+            routeMarkers.Add(routeMarker);
+
+            if (stops.Count > StopCount)
+                Destroy(stops.Dequeue());
```

Comment:
Needs brackets.

Reply:
This doesn't relate to the touch controls added in this PR and can be roped into the General Samples Code Cleanup issue.